pcap/pcap.go

```diff
@@ -127,11 +150,16 @@ const bpfInstructionBufferSize = 8 * MaxBpfInstructions
 // Handles are already pcap_activate'd
 type Handle struct {
 	// cptr is the handle for the actual pcap C object.
-	cptr         *C.pcap_t
-	blockForever bool
-	device       string
-	deviceIndex  int
-	mu           sync.Mutex
+	cptr        *C.pcap_t
+	timeout     time.Duration
+	device      string
+	deviceIndex int
+	mu          sync.Mutex
+	closeMu     sync.Mutex
+	// stop is closed by Handle.Close to signal to getNextBufPtrLocked to stop
+	// trying to read packets
+	stop chan bool
```

Comment:
Instead of a stop channel here, let's use an atomic.  From quick benchmarking, they appear to be around 4x faster:

```go
package main

import (
        "sync/atomic"
        "testing"
)

func BenchmarkSelectBool(b *testing.B) {
        c := make(chan bool)
        for i := 0; i < b.N; i++ {
                select {
                case <-c:
                default:
                }
        }
}
func BenchmarkSelectStruct(b *testing.B) {
        c := make(chan struct{})
        for i := 0; i < b.N; i++ {
                select {
                case <-c:
                default:
                }
        }
}
func BenchmarkAtomic(b *testing.B) {
        x := uint64(0)
        for i := 0; i < b.N; i++ {
                if atomic.LoadUint64(&x) != 0 {
                        b.Fatal("oops")
                }
        }
}
```

gives this result:

```sh
gconnell@rockyduck:~$ go test select_test.go --test.bench=.
testing: warning: no tests to run
BenchmarkSelectBool-12          200000000                9.10 ns/op
BenchmarkSelectStruct-12        200000000                9.10 ns/op
BenchmarkAtomic-12              1000000000               2.51 ns/op
PASS
ok      command-line-arguments  8.305s
```